### PR TITLE
feat(DFD-415): update DS's badge & popover components

### PR DESCRIPTION
**Pull request by jmainguytalend**
**What is the problem this PR is trying to solve?**

- Popover typings is wrong.
- Implement Badge popover correctly.

**What is the chosen solution to this problem?**

Update popover typings and badge popover implementation.

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/design-system/src/components/Badge/primitive/BadgePrimitive.tsx [hunk 1]
```diff
@@ -31,7 +32,6 @@ export interface BadgeDropdownItem {
 export interface BadgePopoverItem {
 	id: string;
 	label: string;
-	onClick: () => void;
```
Comment (aamalric-talend) [inline packages/design-system/src/components/Badge/primitive/BadgePrimitive.tsx:34]: 👍

File: packages/design-system/src/components/Badge/variants/BadgePopover.tsx [hunk 2]
```diff
@@ -1,42 +1,52 @@
...
-							</BadgeButton>
-						</Fragment>
-					))}
-				</StackHorizontal>
-			}
+			<Popover
+				disclosure={
+					<BadgeButton
+						componentId={'badgepopover-button'}
+						data-testid={'badgepopover-button'}
```
Comment (aamalric-talend) [inline packages/design-system/src/components/Badge/variants/BadgePopover.tsx:30]: You lose the feature of data-test customization here compared to previous code right ?

File: packages/design-system/src/stories/messaging/Popover.stories.tsx [hunk 3]
```diff
@@ -55,7 +54,6 @@ export const FormDisclosureStory = () => (
 	<div style={{ padding: '0.75rem' }}>
 		<Popover
 			aria-label="Custom popover"
-			focusOnDisclosure
```
Comment (aamalric-talend) [inline packages/design-system/src/stories/messaging/Popover.stories.tsx:58]: Is this removal on purpose ?

File: packages/design-system/src/components/Badge/variants/BadgePopover.tsx (hunk 2 above)
Comment (jmainguytalend) [reply packages/design-system/src/components/Badge/variants/BadgePopover.tsx:30]: Yes. Before we have button per element in "value" but we want only one button with all element in value as label of it.

File: packages/design-system/src/stories/messaging/Popover.stories.tsx (hunk 3 above)
Comment (jmainguytalend) [reply packages/design-system/src/stories/messaging/Popover.stories.tsx:58]: Props doesn't exist anymore in Popover component

File: packages/design-system/src/components/Badge/variants/BadgePopover.tsx [hunk 4]
```diff
@@ -1,42 +1,52 @@
...
+		value: BadgePopoverItem[];
+	}
+>;
 
 const BadgePopover = forwardRef((props: BadgePopoverProps, ref: Ref<HTMLSpanElement>) => {
-	const { value } = props;
+	const { children, value } = props;
+
+	const [showPopover, setShowPopover] = useState(false);
+	// const popover = usePopover({ modal: false, open: showPopover, placement: 'bottom-start' });
```
Comment (tlnd-mhuchet) [inline packages/design-system/src/components/Badge/variants/BadgePopover.tsx:22]: To remove maybe?

```suggestion
```

Comment (jmainguytalend) [reply packages/design-system/src/components/Badge/variants/BadgePopover.tsx]: Good catch! thank you

**Review by tlnd-mhuchet — APPROVED** [review]
